include/bouffalolab/common/adc_reg.h

```diff
@@ -0,0 +1,268 @@
+
+/* 0x0 : gpadc_config */
+#define GPIP_GPADC_DMA_EN        (1 << 0U)
+#define GPIP_GPADC_FIFO_CLR      (1 << 1U)
+#define GPIP_GPADC_FIFO_NE       (1 << 2U)
+#define GPIP_GPADC_FIFO_FULL     (1 << 3U)
+#define GPIP_GPADC_RDY           (1 << 4U)
+#define GPIP_GPADC_FIFO_OVERRUN  (1 << 5U)
+#define GPIP_GPADC_FIFO_UNDERRUN (1 << 6U)
+#if defined(BL702) || defined(BL702L)
+#define GPIP_GPADC_FIFO_RDY (1 << 7U)
+#endif
+#define GPIP_GPADC_RDY_CLR            (1 << 8U)
+#define GPIP_GPADC_FIFO_OVERRUN_CLR   (1 << 9U)
+#define GPIP_GPADC_FIFO_UNDERRUN_CLR  (1 << 10U)
+#define GPIP_GPADC_RDY_MASK           (1 << 12U)
+#define GPIP_GPADC_FIFO_OVERRUN_MASK  (1 << 13U)
+#define GPIP_GPADC_FIFO_UNDERRUN_MASK (1 << 14U)
+#if defined(BL702) || defined(BL702L)
+#define GPIP_GPADC_FIFO_RDY_MASK (1 << 15U)
+#endif
+#define GPIP_GPADC_FIFO_DATA_COUNT_SHIFT (16U)
+#define GPIP_GPADC_FIFO_DATA_COUNT_MASK  (0x3f << GPIP_GPADC_FIFO_DATA_COUNT_SHIFT)
+#define GPIP_GPADC_FIFO_THL_SHIFT        (22U)
+#define GPIP_GPADC_FIFO_THL_MASK         (0x3 << GPIP_GPADC_FIFO_THL_SHIFT)
+
+/* 0x4 : gpadc_dma_rdata */
+#define GPIP_GPADC_DMA_RDATA_SHIFT (0U)
+#define GPIP_GPADC_DMA_RDATA_MASK  (0x3ffffff << GPIP_GPADC_DMA_RDATA_SHIFT)
+
+/* 0x20 : gpadc_pir_train */
+#define GPIP_PIR_EXTEND_SHIFT (0U)
+#define GPIP_PIR_EXTEND_MASK  (0x1f << GPIP_PIR_EXTEND_SHIFT)
+#define GPIP_PIR_CNT_V_SHIFT  (8U)
+#define GPIP_PIR_CNT_V_MASK   (0x1f << GPIP_PIR_CNT_V_SHIFT)
+#define GPIP_PIR_TRAIN        (1 << 16U)
+#define GPIP_PIR_STOP         (1 << 17U)
+
+/* 0x90C : gpadc_reg_cmd */
+#define AON_GPADC_GLOBAL_EN     (1 << 0U)
+#define AON_GPADC_CONV_START    (1 << 1U)
+#define AON_GPADC_SOFT_RST      (1 << 2U)
+#define AON_GPADC_NEG_SEL_SHIFT (3U)
+#define AON_GPADC_NEG_SEL_MASK  (0x1f << AON_GPADC_NEG_SEL_SHIFT)
+#define AON_GPADC_POS_SEL_SHIFT (8U)
+#define AON_GPADC_POS_SEL_MASK  (0x1f << AON_GPADC_POS_SEL_SHIFT)
+#define AON_GPADC_NEG_GND       (1 << 13U)
+#define AON_GPADC_MICBIAS_EN    (1 << 14U)
+#define AON_GPADC_MICPGA_EN     (1 << 15U)
+#define AON_GPADC_BYP_MICBOOST  (1 << 16U)
+#if defined(BL616) || defined(BL606P) || defined(BL808) || defined(BL628)
+#define AON_GPADC_RCAL_EN (1 << 17U)
+#endif
+#define AON_GPADC_DWA_EN              (1 << 18U)
+#define AON_GPADC_MIC2_DIFF           (1 << 19U)
+#define AON_GPADC_MIC1_DIFF           (1 << 20U)
+#define AON_GPADC_MIC_PGA2_GAIN_SHIFT (21U)
+#define AON_GPADC_MIC_PGA2_GAIN_MASK  (0x3 << AON_GPADC_MIC_PGA2_GAIN_SHIFT)
+#define AON_GPADC_MICBOOST_32DB_EN    (1 << 23U)
+#define AON_GPADC_CHIP_SEN_PU         (1 << 27U)
+#define AON_GPADC_SEN_SEL_SHIFT       (28U)
+#if defined(BL616) || defined(BL606P) || defined(BL808) || defined(BL628)
+#define AON_GPADC_SEN_SEL_MASK (0x7 << AON_GPADC_SEN_SEL_SHIFT)
+#define AON_GPADC_SEN_TEST_EN  (1 << 31U)
+#elif defined(BL702) || defined(BL602) || defined(BL702L)
+#define AON_GPADC_SEN_SEL_MASK (0x3 << AON_GPADC_SEN_SEL_SHIFT)
+#define AON_GPADC_SEN_TEST_EN  (1 << 30U)
+#endif
```

Comment:
This is a common file but have specifics, so not common anymore.
Non blocking at moment but this should be moved to `include/bouffalolab/<soc>/foo_reg.h`

Maybe in future we work with auto-generators to avoid this kind of problems.
With that, we should use enums and make compiler warn when using an invalid field.

Reply:
Ill try and scan all the common files for similar things, given we dont use the BLxxx defines anymore this could be a problem later.

Reply:
Done, under testing.

Reply:
Ok, looks good, but i'm 100 % expecting to have this change this later.